core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceCloneSpec.scala

```diff
@@ -0,0 +1,1398 @@
+    val billingProfileId = UUID.randomUUID()
+    val billingProject = RawlsBillingProject(
+      RawlsBillingProjectName(destWorkspaceRequest.namespace),
+      CreationStatuses.Ready,
+      None,
+      None,
+      billingProfileId = Some(billingProfileId.toString)
+    )
+    val billingProfile = mock[ProfileModel]
+    when(billingProfile.getCloudPlatform).thenReturn(CloudPlatform.AZURE)
+    val billingProfileManagerDAO = mock[BillingProfileManagerDAO]
+    when(billingProfileManagerDAO.getBillingProfile(billingProfileId, testContext)).thenReturn(Some(billingProfile))
+    val samDAO = mock[SamDAO]
+    when(samDAO.getUserStatus(testContext)).thenReturn(Future(Some(SamUserStatusResponse("", "", true))))
+    when(
+      samDAO.userHasAction(
+        SamResourceTypeNames.workspace,
+        sourceWorkspace.workspaceId,
+        SamWorkspaceActions.read,
+        testContext
+      )
+    ).thenReturn(Future(true))
+    when(
+      samDAO.userHasAction(
+        SamResourceTypeNames.billingProject,
+        billingProject.projectName.value,
+        SamBillingProjectActions.createWorkspace,
+        testContext
+      )
+    ).thenReturn(Future(true))
+    val workspaceRepository = mock[WorkspaceRepository]
+    when(workspaceRepository.getWorkspace(sourceWorkspace.toWorkspaceName, None))
+      .thenReturn(Future(Some(sourceWorkspace)))
+    val billingRepository = mock[BillingRepository]
+    when(billingRepository.getBillingProject(RawlsBillingProjectName(destWorkspaceRequest.namespace)))
+      .thenReturn(Future(Some(billingProject)))
+    val service = spy(
+      new MultiCloudWorkspaceService(
+        testContext,
+        mock[WorkspaceManagerDAO],
+        billingProfileManagerDAO,
+        samDAO,
+        mock[MultiCloudWorkspaceConfig],
+        mock[LeonardoDAO],
+        "MultiCloudWorkspaceService-test",
+        mock[WorkspaceManagerResourceMonitorRecordDao],
+        workspaceRepository,
+        billingRepository
+      )
+    )
+    val destWorkspace = mock[Workspace]
+    doReturn(Future(destWorkspace))
+      .when(service)
+      .cloneAzureWorkspaceAsync(sourceWorkspace, billingProfile, destWorkspaceRequest, testContext)
+
+    val result = Await.result(
+      service.cloneMultiCloudWorkspaceAsync(
+        mock[WorkspaceService],
+        sourceWorkspace.toWorkspaceName,
+        destWorkspaceRequest
+      ),
+      Duration.Inf
+    )
+
+    result shouldBe WorkspaceDetails.fromWorkspaceAndOptions(destWorkspace,
+                                                             Some(Set.empty),
+                                                             useAttributes = true,
+                                                             Some(WorkspaceCloudPlatform.Azure)
+    )
+    verify(service).cloneAzureWorkspaceAsync(sourceWorkspace, billingProfile, destWorkspaceRequest, testContext)
+  }
+
+  it should "pass a request to clone a GCP workspace to cloneWorkspace in workspaceService" in {
```

Comment:
The description of this seems the same as https://github.com/broadinstitute/rawls/pull/2939/files#diff-6a2f47cec9c431e15a9b8800cc081a8a558d119d8c0f952ad655263c6f6af621R338, but this one does have an auth domain. I'm wondering if the tests should be combined, or if the test description should be modified to make it clear what is different about this one.

Reply:
The difference is that one is for the sync version (`cloneMultiCloudWorkspace`), and one is for the async version (`cloneMultiCloudWorkspaceAsync`), and that part of the functionality should be the same for both versions.

The output on the test report is probably a bit more clear, because the grouping is more obvious.